keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1086 @@
+      fields)
+    - Allow pods that set new fields that the policy level has an opinion about if the value is the
+      default (explicit or implicit) value OR the value is allowed by newer versions of the policy
+      level (e.g. a less privileged value of a new field)
+- Under the webhook implementation, policy versions are tied to the webhook version, not the cluster
+  version. This means that it is recommended for the webhook to updated prior to updating the
+  cluster. Note that policies are not guaranteed to be backwards compatible, and a newer restricted
+  policy could require setting a field that doesn't exist in the current API version.
+
+For example, the restricted policy level now requires `allowPrivilegeEscalation=false`, but this
+field wasn't added until Kubernetes v1.8, and all containers prior to v1.8 implicitly ran as
+`allowPrivilegeEscalation=true`. Under the **restricted v1.7** profile, the following
+`allowPrivilegeEscalation` configurations would be allowed on a v1.8 cluster:
+- `null` (allowed during the v1.7 release)
+- `true` (equal in privilege to a v1.7 pod that didn't set the field)
+- `false` (strictly less privileged than other allowed values)
+
+<<[UNRESOLVED]>>
+
+_Blocking for Beta._
+
+How long will old profiles be kept for? What is the removal policy?
+
+<<[/UNRESOLVED]>>
+
+### PodTemplate Resources
+
+Audit and Warn modes are also checked on resource types that embed a PodTemplate (enumerated below),
+but enforce mode only applies to actual pod resources.
+
+Since users do not create pods directly in the typical deployment model, the warning mechanism is
+only effective if it can also warn on templated pod resources. Similarly, for audit it is useful to
+tie the audited violation back to the requesting user, so audit will also apply to templated pod
+resources. In the interest of supporting mutating admission controllers, policies will only
+be enforced on actual pods.
+
+Templated pod resources include:
+
+- v1 ReplicationController
+- v1 PodTemplate
+- apps/v1 ReplicaSet
+- apps/v1 Deployment
+- apps/v1 StatefulSet
+- apps/v1 DaemonSet
+- batch/v1 CronJob
+- batch/v1 Job
+
+PodTemplate warnings & audit will only be applied to built-in types. CRDs that wish to take
+advantage of this functionality can use an object reference to a v1/PodTemplate resource rather than
+inlining a PodTemplate. We will publish a guide (documentation and/or examples) that demonstrate
+this pattern. Alternatively, the functionality can be implemented in a 3rd party admission plugin
+leveraging the library implementation.
+
+### Namespace policy update warnings
+
+When an `enforce` policy (or version) label is added or changed, the admission plugin will test each pod
+in the namespace against the new policy. Violations are returned to the user as warnings. These
+checks have a timeout of XX seconds and a limit of YY pods, and will return a warning in the event
```

Comment:
In a large namespaces, I imagine someone running a dry-run enforce, having these limits expire before we find violating pods (even though they exist), user flipping on enforcement, and stuff breaking. Is that a concern?